src/components/RecommendationPage.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import axiosInstance from '../apis/axiosClient';
+import { Typography, CircularProgress, List, ListItem } from '@mui/material';
+import { useAuth } from '@akosasante/react-auth-context';
+const RecommendationPage = () => {
+  const { user } = useAuth();
+  const userId = user ? user.userId : null;
```

Comment:
Please take a look at my previous comment about how `user` is nested and it does not have a `userId` field. If you run `console.log` or add a breakpoint in the browser on this line, you will see what I mean.

This needs to be:

```suggestion
  const userId = (user && user.user) ? user.user._id : null;
```

or even shorter, with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining):

```javascript
const userId = user?.user?._id
```

Reply:
I have removed user  in line 20 in App.js commit 94f2987a8f8c26ab2828a4c628366bde8a193ea6